sbi/simulators/simutils.py

```diff
@@ -48,6 +97,7 @@ def simulate_in_batches(
             for batch in batches:
```

Comment:
In general, you are right, one can make this much shorter, e.g. as you suggest with` cat([simulator(b) for b in tqdm(batches)], 0)`. Here is why we do not do this:

1) we want to support `show_progressbar` as string. If the user wants to provide any additional information for the run (e.g. hyperparams or whatever), they can be displayed here. It's a minor point and I am fine with getting rid of it. It also existed in delfi though.
2) if we use `cat([simulator(b) for b in tqdm(batches)], 0)` the progressbar will just range from 0 to number_of_batches, i.e. it will not show the actual number of simulations we have run, but instead the number of batches we have processed.

The syntax is correct, what is your worry there?

Reply:
ok, I see for the showing of simulations (2.) though that is easily reconstructable knowing the simulation batch_size. Agree to leave it as it is. About (1.) my concern stemmed from a typing perspective, `show_progressbar` should be then `Union[str, bool]`. 

In fact, thinking about (2) the statement where you set the description could be used to display the simulation batch_size. 

Overall, I have a bias against code that gets deeply nested and complicated not for algorithmic reasons but because of 'paraphernalia' (as it gets more difficult to read by somebody not initiated in progress bars - I'd like that the precondition to read our code be rather understanding of the algorithms that understanding of the capricious ways of some logging library).